src/libecalc/common/graph.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from typing import Dict, Generic, List, Protocol, TypeVar
+
+import networkx as nx
+
+NodeID = str
+
+
+class NodeWithID(Protocol):
+    @property
+    def id(self) -> NodeID:
+        ...
+
+
+TNode = TypeVar("TNode", bound=NodeWithID)
+
+
+class Graph(Generic[TNode]):
```

Comment:
You _may_be_ right that I want more domain specific names, but perhaps not here? Depends on how we use this graph class. If this graph class is never exposed to users, but just an internal interface because we use a "graph-like structure", then it is ok, but that the implementation (networkx) is hidden.

However, in that case this is a data structure that could be "outside" the domain, in a "helper library, such as seedwork that is used similarily as a part of the language", such as pydantic , since Python itself doesnt have a graph-like structure in its batteries (afaik)